packages/lpa-questionnaire-web-app/src/controllers/upload-plans.js

```diff
@@ -0,0 +1,104 @@
+const { VIEW } = require('../lib/views');
+const {
+  fileErrorSummary,
+  fileUploadNunjucksVariables,
+  deleteFile,
+  uploadFiles,
+} = require('../lib/file-upload-helpers');
+const getAppealSideBarDetails = require('../lib/appeal-sidebar-details');
+const { getTaskStatus } = require('../services/task.service');
+const { createOrUpdateAppealReply } = require('../lib/appeal-reply-api-wrapper');
+
+exports.getUploadPlans = (req, res) => {
+  const { uploadedFiles = [] } = req.session.appealReply.requiredDocumentsSection.plansDecision;
+
+  // set uploaded files
+  req.session.uploadedFiles = uploadedFiles;
+
+  res.render(VIEW.UPLOAD_PLANS, {
+    appeal: getAppealSideBarDetails(req.session.appeal),
+    backLink: req.session.backLink || `/${req.params.id}/${VIEW.TASK_LIST}`,
+    ...fileUploadNunjucksVariables(null, null, uploadedFiles),
+  });
+};
+
+exports.postUploadPlans = async (req, res) => {
+  const documents = req.body?.files?.documents || [];
+
+  const { delete: deleteId = '', errors = {}, submit = '' } = req.body;
+
+  const backLink = req.session.backLink || `/${req.params.id}/${VIEW.TASK_LIST}`;
+
+  let errorMessage;
+
+  // Chance for delete to be triggered due to non-JS solution. delete will be set to value of filename if button clicked
+  if (deleteId) {
+    try {
+      await deleteFile(deleteId, req);
+    } catch (err) {
+      req.log.error({ err }, `Error deleting ${deleteId} from Upload Plans`);
+    }
+  } else if (documents.length) {
+    // Chance for files to be attached due to non-JS solution, these need to be uploaded (with appropriate errors);
+    try {
+      const uploadedFiles = await uploadFiles(
+        documents.map((doc, index) => ({
+          ...doc,
+          // current validation will return errors in this format
+          error: errors[`files.documents[${index}]`]?.msg,
+        })),
+        req.session?.appealReply?.id
```

Comment:
I think we should abort early if `req.session.appealReply.id` is undefined. With this approach we are still at risk of placing documents into the `undefined` blob in the document store, which will have impact down the line when needing to retrieve data.

Reply:
Although technically possible at this stage, with the changes coming it will shortly be impossible to get to this stage without the ID being present. This eliminates the risk. MVP for questionnaire includes that piece of work

Reply:
Fair enough - I see approval already on the ticket.

My personal opinion is that I'm reviewing this as the code stands now, and there is a risk here that is being accepted. I'm not comfortable with that risk, but if those directly involved in the ticket are happy to proceed then by all means disregard this feedback.

Reply:
I appreciate where you are coming from. The only chance of the ID not being set even in this version is if the Appeal Reply service is down for some reason.

It might be we can add a check for the appealReply existence in session, but would need to add that to every question ideally (at the moment it is assumed to be there). We took that from Team 1 work so might need to think wider on it